src/main/java/org/opensearch/neuralsearch/processor/factory/NormalizationProcessorFactory.java

```diff
@@ -73,6 +75,11 @@ public SearchPhaseResultsProcessor create(
                 TECHNIQUE,
                 ArithmeticMeanScoreCombinationTechnique.TECHNIQUE_NAME
             );
+            // case when technique is z score and combination is not arithmetic mean
+            if (normalizationTechniqueName.equals(ZScoreNormalizationTechnique.TECHNIQUE_NAME)
```

Comment:
Can you extract this validation out of this method to a separate one? Also add why we are blocking these techniques?